Quadriga/src/main/java/edu/asu/spring/quadriga/web/settings/WebsiteAboutEditController.java

```diff
@@ -88,19 +111,60 @@ public String editAbout(@ProjectIdentifier @PathVariable("ProjectId") String pro
-            BindingResult result, ModelAndView model, Principal principal) throws QuadrigaStorageException {
+            @Validated @ModelAttribute("aboutTextBean") AboutText formBean, BindingResult result, ModelAndView model,
+            Principal principal, @InjectProject IProject project) throws QuadrigaStorageException {
         model = new ModelAndView("auth/editabout");
+        List<INetwork> networks = nwManager.getNetworksInProject(projectId, INetworkStatus.APPROVED);
         model.addObject("project", project);
+        model.addObject("networks", networks);
         if (result.hasErrors()) {
             model.addObject("aboutTextBean", formBean);
         } else {
             aboutTextManager.saveAbout(projectId, formBean);
             model.addObject("show_success_alert", true);
             model.addObject("success_alert_msg", "You successfully edited the about text");
-
         }
         return model;
     }
 
-}
+    /**
+     * This method is used to return a JSON string for visualizing a network
+     * based on the Network id selected from the UI.
+     * 
+     * @param projectId
+     *            Unix Name given for the Project
+     * @param networkId
+     *            Network Id for the network selected from the UI.
+     * @param principal
+     *            principal object which is required to fetch information about
+     *            logged in user.
+     * 
+     * @return Returns a JSON string as a response entity based on the network
+     *         selected from the UI.
+     * @throws QuadrigaStorageException
+     * @throws JAXBException
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 4, userRole = {
+            RoleNames.ROLE_COLLABORATOR_OWNER, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/projects/{projectId}/settings/editabout/visualize/{networkId}", method = RequestMethod.GET, produces = "text/plain")
+    @InjectProjectById
+    public ResponseEntity<String> visualizeNetworks(@ProjectIdentifier @PathVariable("projectId") String projectid,
+            @PathVariable("networkId") String networkId, Principal principal,
+            @CheckAccess @InjectProject IProject project) {
+        ITransformedNetwork transformedNetwork = null;
+        try {
+            transformedNetwork = transformationManager.getTransformedNetwork(networkId);
+        } catch (QuadrigaStorageException qse) {
+            logger.error("Error while retrieving networks for display:", qse);
+            return new ResponseEntity<String>("", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        String json = null;
+        if (transformedNetwork != null) {
+            json = jsonCreator.getJson(transformedNetwork.getNodes(), transformedNetwork.getLinks());
+        } else {
+            return new ResponseEntity<String>(json, HttpStatus.NOT_FOUND);
+        }
+
+        return new ResponseEntity<String>(json, HttpStatus.OK);
```

Comment:
@jdamerow The access policies and the path are different. Should I write a utility class and replace the code here?